feature-gate/program/src/instruction.rs

```diff
@@ -0,0 +1,87 @@
+//! Program instructions
+
+use {
+    borsh::{BorshDeserialize, BorshSchema, BorshSerialize},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        system_program,
+    },
+};
+
+/// Feature Gate program instructions
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq)]
+pub enum FeatureGateInstruction {
+    /// Submit a feature for activation.
+    ///
+    /// Accounts expected by this instruction:
```

Comment:
Let's make it clear that this doesn't fund the rent-exempt minimum, and isn't required since you can just create the account separately.

Unless I missed some conversations about how this should work